lib/checks/navigation/heading-order-evaluate.js

```diff
@@ -1,26 +1,42 @@
 import cache from '../../core/base/cache';
 import { querySelectorAllFilter, getAncestry } from '../../core/utils';
 import { isVisible } from '../../commons/dom';
+import { getRole } from '../../commons/aria';
 
 function getLevel(vNode) {
-  const role = vNode.attr('role');
-  if (role && role.includes('heading')) {
-    const ariaHeadingLevel = vNode.attr('aria-level');
-    const level = parseInt(ariaHeadingLevel, 10);
-
-    // default aria-level for a heading is 2 if it is
-    // not set or set to an incorrect value
-    // @see https://www.w3.org/TR/wai-aria-1.1/#heading
-    if (isNaN(level) || level < 1 || level > 6) {
-      return 2;
-    }
+  const role = getRole(vNode);
+  const headingRole = role && role.includes('heading');
+  const ariaHeadingLevel = vNode.attr('aria-level');
+  const ariaLevel = parseInt(ariaHeadingLevel, 10);
+
+  const [, headingLevel] = vNode.props.nodeName.match(/h(\d)/) || [];
+
+  if (!headingRole) {
+    return -1;
+  }
 
-    return level;
+  if (headingLevel && !ariaHeadingLevel) {
+    return parseInt(headingLevel, 10);
   }
 
-  const headingLevel = vNode.props.nodeName.match(/h(\d)/);
-  if (headingLevel) {
-    return parseInt(headingLevel[1], 10);
+  /*
+   * default aria-level for a role=heading is 2 if it is
+   * not set or set to an incorrect value. 
+   * default aria-level for a heading element is the 
+   * heading level.
+   * note that NVDA and VO allow any positive level
+   * @see https://www.w3.org/TR/wai-aria-1.1/#heading
+   * @see https://codepen.io/straker/pen/jOBjNNe
+   */
+  if (isNaN(ariaLevel) || ariaLevel < 1 || ariaLevel > 6) {
```

Comment:
Alright. Wilco and I talked and we came to the conclusion that we want to change how we're handling positive values. Since `aria-level` > 6 is handled different by different screen readers, we're going to flag it as part of a new check for aria-valid-attr-value and call out its lack of support.

But for heading-order we're just going to accept what the user gave us so there isn't two different failures caused by the same problem.

```suggestion
  if (isNaN(ariaLevel) || ariaLevel < 1) {
```